docs/cvl/builtin.md

```diff
@@ -1,113 +1,179 @@
 Built-in Rules
 ==============
-The Certora Prover has built-in general-purpose rules targeted at finding known vulnerabilities.
-These rules can be verified on a contract out-of-the-box.
 
-Users can add built-in rules and other rule types in the same spec file.
+The Certora Prover has built-in general-purpose rules targeted at finding common
+vulnerabilities.  These rules can be verified on a contract without writing any
+contract-specific rules.
 
+Built-in rules can be included in any spec file by writing `use builtin rule
+<rule-name>;`.  This document describes the available built-in rules.
+
+```{contents}
+```
 
 Syntax
 ------
-The syntax for using the built-in rule identifier
+
+The syntax for rules is given by the following [EBNF grammar](syntax):
 
 ```
-built-in-rule ::= "use" "builtin" "rule" id ";"
-```
+built_in_rule ::= "use" "builtin" "rule" built_in_rule_name ";"
 
-Example
--------
-In order to run the built-in rule `msgValueInLoopRule` add to the spec file the line
+built_in_rule_name ::=
+    | "msgValueInLoopRule"
+    | "hasDelegateCalls"
+    | "sanity"
+    | "deepSanity"
+```
 
-`use builtin rule msgValueInLoopRule;`
+(built-in-msg-value-in-loop)=
+Bad loop detection &mdash; `msgValueInLoopRule`
+-----------------------------------------------
 
-The Names of Currently Implemented Rules:
------------------------------------------
-- `msgValueInLoopRule` – checks for occurrences of `msg.value` and delegate calls in loops.
-- `hasDelegateCalls` - checks for delegate calls anywhere in the contract.
-- `sanity`
-- `deepSanity`
+Loops that [use `msg.value`][msg-value-vulnerability] or [make delegate
+calls][delegatecall-vulnerability] are a well-known source of security
+vulnerabilities.
 
+[msg-value-vulnerability]: https://trustchain.medium.com/ethereum-msg-value-reuse-vulnerability-5afd0aa2bcef
+[delegatecall-vulnerability]: https://blog.trailofbits.com/2021/12/16/detecting-miso-and-opyns-msg-value-reuse-vulnerability-with-slither/
 
```

Comment:
I would not cite slither...
https://blog.solidityscan.com/security-issues-with-delegate-calls-4ae64d775b76 maybe this one?

Reply:
The link for msg.value mentions the relation to delegate calls so it is enough for the msg.value/delegate calls in loops.
I added the link Ghila gave to hasDelegateCalls as this refers to delegate calls in general and does not mention loops.